gibbon-compiler/src/Packed/FirstOrder/Passes/Codegen.hs

```diff
@@ -401,6 +411,23 @@ codegenTail (LetPrimCallT bnds prm rnds body) ty =
                             [ C.BlockDecl [cdecl| $ty:(codegenTy valTy) $id:valV = *( $ty:(codegenTy valTy) *)($id:cur); |]
                             , C.BlockDecl [cdecl| $ty:(codegenTy CursorTy) $id:curV = ($id:cur) + sizeof( $ty:(codegenTy IntTy) ); |] ]
 
+                 ReadCursor -> let [(next,CursorTy)] = bnds
+                                   [(VarTriv cur)] = rnds in pure
+                               [ C.BlockDecl [cdecl| $ty:(codegenTy CursorTy) $id:next = *($ty:(codegenTy CursorTy) *) ($id:cur); |] ]
+
+                 BoundsCheck -> do
```

Comment:
`BoundsCheck` just travels through L2, L3 and L4 and is transformed into C code here. Exactly same as the [microbenchmark](https://github.com/iu-parfunc/gibbon/blob/master/growing_regions/infinite.c#L120)